File: All Topics/9-PalindromeNumber.ts
const x = 121;

function isPalindrome(x: number): boolean {
  const string = String(x);
  let result = true;
  for (let i = 0; i < Math.floor(string.length / 2); i++) {
    if (i === 1) {
      if (string[0] === "-" || string[0] === "+") {
        return (result = false);
      }
    }
    if (string[i] !== string[string.length - i - 1]) {
      return (result = false);
    }
  }

  return result;
}

console.log(isPalindrome(x));

// function isPalindrome(x: number): boolean {

//   return x.toString() === x.toString().split("").reverse().join("")
//     ? true
//     : false;
// }

File: All-Topics/1-50/9-PalindromeNumber.cpp



class Solution {
public:
    bool isPalindrome(int x) {

        if (x < 0) {
            return false;
        }

    
    st::string stringX = std::to_string(x);

    for(int i=0;i<stringX.size();i++){
        if(stringX[i]!=stringX[stringX.size()-i-1])return false;
    }

    return true;


    }
};